src/App.js

```diff
@@ -94,148 +108,177 @@ function App() {
+      <div className="header">
         <div>Skóre: {getScore(history)}</div>
         <div>Městle</div>
         <div>{new Date().toLocaleDateString("cz-CS")}</div>
       </div>
-      <div className='body'>
-        <div className='body-background' style={{ backgroundImage: `url(${background})`}}></div>
+      <div className="body">
+        <div
+          className="body-background"
+          style={{ backgroundImage: `url(${background})` }}
+        ></div>
         <div className="differences title">
           <div className="guess">Kraj</div>
           <div className="guess">Populace</div>
           <div className="guess">Rozloha</div>
           <div className="guess">Nadmořská výška</div>
           <div className="guess">Poloha</div>
         </div>
-        {
-          getGuesses(history, todaySeed).length > 0 &&
+        {getGuesses(history, todaySeed).length > 0 && (
           <>
-            {
-              getGuesses(history, todaySeed).map((g, idx, array) => 
-                <Guess
-                  key={idx}
-                  idx={idx} guessedCity={g}
-                  targetCity={getRandCity(cities, todaySeed)}
-                  isLast={idx === array.length - 1}
-                  isEog={getEog(history, todaySeed)}
-                />
-              )
-            }
+            {getGuesses(history, todaySeed).map((g, idx, array) => (
+              <Guess
+                key={idx}
+                idx={idx}
+                guessedCity={g}
+                targetCity={getRandCity(cities, todaySeed)}
+                isLast={idx === array.length - 1}
+                isEog={getEog(history, todaySeed)}
+              />
+            ))}
           </>
-        }
+        )}
         <div ref={bottom}>&nbsp;</div>
       </div>
-      {
-        !getEog(history, todaySeed) &&
-        <div className='guess-box'>
+      {!getEog(history, todaySeed) && (
+        <div className="guess-box">
+          {filteredCities.length > 0 && (
+            <CitySelector
+              options={filteredCities}
+              selectedOption={cityPart}
+              setSelectedOption={(option) => handleSelectCity(option)}
+            />
+          )}
           <Tippy
-            render={attrs => (
+            render={(attrs) => (
               <Box style={props} {...attrs}>
                 Začni náhodným městem a pak zlepšuj odhad.
               </Box>
             )}
             placement="top"
             theme="orange"
             zIndex={11}
-            disabled={getGuesses(history, todaySeed).length > 0 || cityPart !== ''}
+            disabled={
+              getGuesses(history, todaySeed).length > 0 || cityPart !== ""
+            }
             visible={true}
             animation={true}
             onMount={onMount}
             onHide={onHide}
           >
-            <input value={cityPart} placeholder='Uhádni dněšní město' onChange={event => handleChangeCityPart(event.target.value)}/>
+            <input
```

Comment:
What about to propagate onKeyUp and onKeyDown events from this input to CitySelector?